### PR TITLE
Gun Assembling [Ready for check]

### PR DESCRIPTION
**Pull request by zombiemimic**
## About The Pull Request
Now without merge conflict dumbassery

Adds in framework for assembling firearms from spare parts, along with pistol parts for this.
image
image
image

- [x] pistols
- [x] revolvers
- [x] shotguns
- [x] smgs
- [x] automatic firearms
- [x] rifles

Sprites by @infraredbaron and @shazbot194 

## Why It's Good For The Game
The Gun is Good! Intended as a replacement for the guns found in maintenance in droves.

## Changelog
:cl:
add: Added new gun parts: Frames, Grips, Mechanisms, and Barrels
add: Added new system of building guns from spare parts
tweak: tweaked results of breaking down guns for parts
/:cl:

### CODE REVIEW

**Comment by MLGTASTICa** [comment]
it would be way better if you would use text based recipes instead of hardcoding 4 for each grip. if someone in the future wants to add more they will very much have a hard time and possibly to rework all of the framework.

File: code/datums/craft/gun_parts.dm [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+	var/variant_grip = FALSE
+	var/gripvar1 = /obj/item/part/gun/grip/wood
+	var/gripvar2 = /obj/item/part/gun/grip/black
+	var/gripvar3
+	var/gripvar4
+
+	var/resultvar1 = /obj/item/gun/projectile
+	var/resultvar2 = /obj/item/gun/projectile
+	var/resultvar3
+	var/resultvar4
```
Comment (TheShown911) [inline code/datums/craft/gun_parts.dm:24]: This is really not a good way to do this for different results, you can use a list ordered by indexes or an associative list to change the gun depending on the parts used.

**Comment by drexample** [comment]
Needs another test server run

**Review by drexample — CHANGES_REQUESTED** [review]
What Shown said

**Comment by Sparrowbird611** [comment]
I just want to ask, what exactly is the point of this. The gunpart system is what was meant to be in the trash, not whatever this is. I feel like I'm missing something here.

**Comment by zombiemimic** [comment]
conflicts fixed

**Comment by Sparrowbird611** [comment]
I'm gonna bring it up again since this has been test merged and I never really got a response on my first question. What's the point of this? We have another gunpart system that seems like it could work well if the parts would spawn in maintenance more consistently. Why have this as well?

**Comment by tristain01** [comment]
Alright it has been brought to my attention that this is a replacement for the fire gun part system and seeing as I don't recall seeing the junk guns in the sprite sheets, how will we make them, are they going to be added later?

**Pull request merged**